### PR TITLE
Make various throwing weapons Rogue class and nerf Wulfrum Ammo

### PR DESCRIPTION
**Pull request by CyberGamer64**
As Per the Trello Cards: https://trello.com/c/er7QUJXm and https://trello.com/c/8GWNICQx

The Following Weapons now classified as Rogue:
- Bone Javelin
- Bone Throwing Knife,
- Throwing Knife
- Shuriken
- Soaring Scalpula
- Duskfeather Dagger
- Lightning Throw
- Bismite Grenade
- Kunai (Spirit Mod one)
- Explosive Rum
- Gel Throwing Axe
- Winged Knife
- Berry Bombs
- Explosive Knife

Additional Changes:
- Wulfrum Scrap Ammo now does 5 damage.

Note: The Gold Chakram remains unchanged.

### CODE REVIEW

**Comment by 1Git2Clone** [comment]
Hello.

The ideas you have are great but do you think the code is easy to navigate? I'm sure it can be refactored in order to make it easier to navigate like storing the damage class string literals in a utility module as well as using [guard clauses](https://en.m.wikipedia.org/wiki/Guard_(computer_science) "Wikipedia source") instead of nesting if statements.

> [!NOTE]
> I'm not a maintainer so I can't edit the code directly but that's all I'd suggest.

If you have any questions feel free to ask them.

**Comment by CyberGamer64** [comment]
I have rewrote the Rogue classifications by using lists and a foreach loop as well as implementing Guard Clauses as recommended. Hopefully these changes makes the code more readable and sustainable long term.

However I have been struggling with figuring out a way to get the Rogue damage class outside of using CalamityMod.Find. Can you store something like CalamityMod.Find<DamageClass>("RogueDamageClass") as a Variable? I know you can store the "RogueDamageClass"  as string but I don't think item.DamageType itself wants a string.

File: Common/Global/GlobalVanillaItem.cs [hunk 1]
```diff
@@ -3,39 +3,40 @@
...
+
+		// Changes following Vanilla Weapons To Rogue, uses Integer due to how Vanilla ID work
+		// In order its Bone Javelin, Throwing Knife, Shuriken and Bone Dagger
+		private static List<int> Vanilla_Weapons_To_Rogue = new List<int>
+		{
+			3378,
+			279,
+			42,
+			3379
+		};
```
[the remark this answers is not in the document]
Comment (1Git2Clone) [reply Common/Global/GlobalVanillaItem.cs:24]: Also, you used a mix-up between `PascalCase` and `snake_case` as the naming convention everywhere. Not that there's something inherently wrong with it (although it doesn't follow the language convention which is pretty inconvenient). The naming convention for fields is `camelCase` and the naming convention for `const` fields is `PascalCase`.

[.NET Style Guidelines](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) (point 3 for fields and point 12 for constants)

[the remark this answers is not in the document]
Comment (1Git2Clone) [reply Common/Global/GlobalVanillaItem.cs:24]: I guess I could make a PR with a CI/CD pipeline and an `.editorconfig` file to catch those style errors. It'd be nice to be able to apply them on the spot.

[the remark this answers is not in the document]
Comment (CyberGamer64) [reply Common/Global/GlobalVanillaItem.cs]: That might be nice since its always better to catch stuff like that earlier than later.

**Comment by 1Git2Clone** [comment]
> I have rewrote the Rogue classifications by using lists and a foreach loop as well as implementing Guard Clauses as recommended. Hopefully these changes makes the code more readable and sustainable long term.
> 
> However I have been struggling with figuring out a way to get the Rogue damage class outside of using CalamityMod.Find. Can you store something like CalamityMod.Find("RogueDamageClass") as a Variable? I know you can store the "RogueDamageClass" as string but I don't think item.DamageType itself wants a string.

One possible way to do it is to store it as a `public static readonly` variable in a utility `class` (contained in a utility `namespace`) and access it from there. You also get the performance benefit for not having to call `CalamityMod.Find()` more than once at least for `"RougeDamageClass"`. And you can `[JITWhenModsEnabled("CalamityMod")]` on that static field.

File: Common/Global/GlobalVanillaItem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (1Git2Clone) [reply Common/Global/GlobalVanillaItem.cs]: CI/CD is impossible since it relies on having Terraria installed and having the game files public which is illegal I'm pretty sure. Managed to use a code analysis tool though so it's all good! I won't open my PR since it may have conflicts with your files but you can check the [branch from my fork](https://github.com/1Git2Clone/TheBereftSouls/tree/features/code-analysis) if you're interested.

What's left is to make a `CONTRIBUTING.md` file but that deserves a different branch and PR related to it. Basically you can format with `dotnet format` and you need to check your code style with `dotnet build --no-incremental`.

File: Common/Global/GlobalVanillaItem.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+			{
+				return;
+			}
+			// Uses list to change all listed weapons to Rogue
+			foreach (int vanillaWeaponId in VanillaWeaponsToRogue)
+			{
+				if (item.type == vanillaWeaponId)
+				{
+					item.DamageType = GetModClass.rogueDamageClass;
+					base.SetDefaults(item);
```
Comment (1Git2Clone) [inline Common/Global/GlobalVanillaItem.cs:40]: Gets called 4 times when it's in the foreach loop.

Comment (1Git2Clone) [reply Common/Global/GlobalVanillaItem.cs:40]: > Gets called 4 times when it's in the foreach loop.

Well, my bad, not 4 times because of the if condition but writing it outside the for loop is still easier to understand.

File: Localization/en-US_Mods.TheBereftSouls.hjson
Comment (1Git2Clone) [inline Localization/en-US_Mods.TheBereftSouls.hjson:1]: *Unnecessary change for the PR*

File: Common/Global/GlobalSpiritModItem.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			{
+				return;
+			}
+			// Uses list to change all listed weapons to Rogue
+			foreach (string spiritWeaponName in SpiritWeaponsToRogue)
+			{
+				if (item.type == SpiritMod.Find<ModItem>(spiritWeaponName).Type)
+				{
+					item.DamageType = GetModClass.rogueDamageClass;
+					base.SetDefaults(item);
```
Comment (1Git2Clone) [inline Common/Global/GlobalSpiritModItem.cs:41]: Same as [this comment](https://github.com/bromeex/TheBereftSouls/pull/9/files#r1793423317 "#9 Common/Global/GlobalVanillaItem.cs Line 39 commit: eef2c06d9cbb21c3fb344de9393cc472519491b0 (\"Test Commit
\")").

File: Common/Global/GlobalSOTSItem.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+			{
+				return;
+			}
+			// Uses list to change all listed weapons to Rogue
+			foreach (string SOTSWeaponName in SOTSWeaponsToRogue)
+			{
+				if (item.type == SOTS.Find<ModItem>(SOTSWeaponName).Type)
+				{
+					item.DamageType = GetModClass.rogueDamageClass;
+					base.SetDefaults(item);
```
Comment (1Git2Clone) [inline Common/Global/GlobalSOTSItem.cs:40]: Same as [this comment](https://github.com/bromeex/TheBereftSouls/pull/9/files#r1793423317 "#9 Common/Global/GlobalVanillaItem.cs Line 39 commit: eef2c06d9cbb21c3fb344de9393cc472519491b0 (\"Test Commit
\")").
